Content.Shared/Random/WeightedRandomPrototype.cs

```diff
@@ -12,4 +12,7 @@ public sealed class WeightedRandomPrototype : IPrototype
 
     [DataField("weights")]
     public Dictionary<string, float> Weights = new();
+
+    [DataField("quantities")]
+    public Dictionary<string, int> Quantities = new();
```

Comment:
is there a reason you need to add this to WeightedRandom? This feels oddly specific for what this prototype is meant to be. This could easily be on the component for `SolutionRandomFill`

Reply:
i've moved all functionality to `SolutionRandomFill`, the `weightedRandom` system should be unchanged by this PR now